bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java

```diff
@@ -41,6 +45,11 @@ protected synchronized void modified(Map<String, Object> config) {
         final String script = (String) config.get(SCRIPT);
         final String region = (String) config.get(REGION);
         final String variant = (String) config.get(VARIANT);
+        final String location = (String) config.get(LOCATION);
+
+        if (location != null) {
+            this.location = PointType.valueOf(location);
```

Comment:
I agree, indeed its a problem if the user provides a mal-formed String. I will catch the `IllegalArgumentException` and keep the old value or `null`, i.e. whatever was set before the wrongly entered information.